toss/fitness/fitness_function_utils.py

```diff
@@ -143,25 +117,140 @@ def compute_space_coverage(number_of_spacecrafts: int, spin_axis: np.ndarray, sp
+        ratio_unique_visits = unique_visits[0].shape[0] / bool_tensor.size
 
-        # Define a zero-valued tensor with the same shape as the spherical meshgrid
-        r_idx = np.where(bool_tensor == True)[0]
+        # Get weights corresponding to the uniquely visited regions. 
+        r_idx = unique_visits[0]
         sum_of_weights = np.sum(1/r[r_idx])
 
         # Return fitness
-        fitness = ratio + sum_of_weights
-
+        fitness = ratio_unique_visits + sum_of_weights
         return fitness
 
 
+def update_spherical_tensor_grid(number_of_spacecrafts: int, spin_axis: np.ndarray, spin_velocity: float, positions: np.ndarray, velocities: np.ndarray, timesteps: np.ndarray, radius_min: float, radius_max: float, r: np.ndarray, theta: np.ndarray, phi: np.ndarray, bool_tensor: np.ndarray) -> np.ndarray:
+    """
+    The function adjusts the trajectory for the body's sidereal rotation, identifies the 
+    positions within the region of interest, matches these positions to the closest points on the
+    spherical meshgrid and update the boolean tensor accordingly.
+
+    Args:
+        number_of_spacecrafts (int): Number of spacecraft
+        spin_axis (np.ndarray): The axis around which the body rotates.
+        spin_velocity (float): Angular velocity of the body's rotation.
+        positions (np.ndarray): (3,N) Array of positions (in cartesian coordinates).
+        velocities (np.ndarray): (3,N) Array of positions (in cartesian frame).
+        timesteps (np.ndarray): (N) Array of time values for each position.
+        radius_min (float): Inner radius of spherical grid, typically radius_inner_bounding_sphere.
+        radius_max (float): Outer radius of spherical grid, typically radius_outer_bounding_sphere.
+        r (np.ndarray): Array of r coordinates for each point defined on the spherical tensor.
+        theta (np.ndarray): Array of theta coordinates for each point defined on the spherical tensor.
+        phi (np.ndarray): Array of phi coordinates for each point defined on the spherical tensor.
+        bool_tensor (np.ndarray): Boolean tensor corresponding to each point defined on the spherical grid.
+
+    Returns:
+        bool_tensor (np.ndarray): Updated boolean tensor corresponding to each point defined on the spherical grid.
+    """
+    # Rotate positions according to body's rotation to simulate that the grid (i.e gravitational field approximation) is also rotating accrdingly
+    rotated_positions = None
+    
+    pos = np.array_split(positions, number_of_spacecrafts, axis=1)
+    for counter, pos_arr in enumerate(pos):
+
+        rot_pos_arr = np.empty((pos_arr.shape))
+
+        for col in range(0,len(pos_arr[0,:])):
+            rot_pos_arr[:,col] = rotate_point(timesteps[col], pos_arr[:,col], spin_axis, spin_velocity)
+
+        if counter == 0:
+            rotated_positions = rot_pos_arr
+        else:
+            rotated_positions = np.hstack((rotated_positions, rot_pos_arr))
```

Comment:
I'm not completely sure on how to vectorize it, but it would be way better if possible. I think this could perhaps be done in another PR as a new feature, especially since it might be a little more complex. The issue is that we will have to create a "rotation object" for each individual angle of rotation as the pyquaternion package can only manage multiple rotations for the same angle. However, perhaps this could be done by creating all of the quaternion objects and converting them to their corresponding rotation matrices, stack these matrices and multiply with the array of positions. But I'm not sure if it would save any significant time. What do you think?